source/docs/software/wpilib-tools/robot-simulation/physics-sim.rst

```diff
@@ -0,0 +1,94 @@
+
+WPILib's Simulation Classes
+---------------------------
+
+The following physics simulation classes are available in WPILib:
+
+- LinearSystemSim, for modeling systems with linear dynamics
+- FlywheelSim
+- DifferentialDrivetrainSim
+- ElevatorSim, which models gravity
+- SingleJointedArmSim, which models gravity
+- BatterySim, which simply estimates battery voltage sag based on drawn currents
+
+All simulation classes (with the exception of the differential drive simulator) inherit from the :code:`LinearSystemSim` class. By default, the dynamics are the linear system dynamics :math:`\mathbf{x}_{k+1} = \mathbf{A}\mathbf{x}_k + \mathbf{B}\mathbf{u}_k`. Subclasses override the :code:`UpdateX(x, u, dt)` method to provide custom, nonlinear dynamics, such as modeling gravity.
+
+Usage in User Code
+------------------
+
+The following is available from the WPILib :code:`elevatorsimulation` `example project <https://github.com/mcm001/allwpilib/tree/state-space-v2/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/elevatorsimulation>`__.
+
+In addition to standard objects such as motors and encoders, we instantiate our elevator simulator using known constants such as carriage mass and gearing reduction. We also instantiate an :code:`EncoderSim`, which sets the distance and rate read by our our :code:`Encoder`.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/mcm001/allwpilib/state-space-v2/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/elevatorsimulation/Robot.java
+         :language: java
+         :lines: 54-61
+         :linenos:
+         :lineno-start: 36
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/mcm001/allwpilib/state-space-v2-stage-2/wpilibcExamples/src/main/cpp/examples/ElevatorSimulation/cpp/Robot.cpp
+         :language: cpp
+         :lines: 58-66
+         :linenos:
+         :lineno-start: 36
+
+Next, :code:`teleopPeriodic`/:code:`TeleopPeriodic` (Java/C++) uses a simple PID control loop to drive our elevator to a setpoint 30 inches off the ground.
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/mcm001/allwpilib/state-space-v2/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/elevatorsimulation/Robot.java
+         :language: java
+         :lines: 83-93
+         :linenos:
+         :lineno-start: 36
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/mcm001/allwpilib/state-space-v2-stage-2/wpilibcExamples/src/main/cpp/examples/ElevatorSimulation/cpp/Robot.cpp
+         :language: cpp
+         :lines: 89-99
+         :linenos:
+         :lineno-start: 36
+
+Next, :code:`simulationPeriodic`/:code:`SimulationPeriodic` (Java/C++) uses the voltage applied to the motor to update the simulated position of the elevator.
```

Comment:
What exactly is "simulationPeriodic", this should probably be explained in small detail.